akka-stream/src/main/scala/akka/stream/javadsl/Source.scala

```diff
@@ -628,6 +628,30 @@ final class Source[+Out, +Mat](delegate: scaladsl.Source[Out, Mat]) extends Grap
     new Source(delegate.prependMat(that)(combinerToScala(matF)))
 
   /**
+   * Provides an alternative that will be consumed if this source completes without any
+   * elements passing by. As soon as the first element comes through this stream, the alternative
+   * will be cancelled.
+   *
+   * Note that this Flow will be materialized together with the [[Source]] and just kept
+   * from producing elements by asserting back-pressure until its time comes or it gets
+   * cancelled.
+   *
+   * On errors the stage is failed regardless of source of the error.
+   *
+   * '''Emits when''' element is available from first stream or first stream closed without emitting any elements and an element
+   *                  is available from the second stream
+   *
+   * '''Backpressures when''' downstream backpressures
+   *
+   * '''Completes when''' the first stream completes after emitting at least one element or else when the second stream completes
+   *
+   * '''Cancels when''' downstream cancels and additionally the alternative is cancelled as soon as an element passes
+   *                    by from this stream.
+   */
+  def orElse[T >: Out, M](alternative: Graph[SourceShape[T], M]): javadsl.Source[T, Mat] =
```

Comment:
explicitly document that the materialized value always comes from the first source not the second. Btw, this can be somewhat awkward, so I am hesitant. The most meaningful would probably be a `Future[Either[M1, M2]]` which only returns once there is a decision on which source to consume, and then returns the materialized value of that one. I admit it is tricky.